quickcheck-dynamic/src/Test/QuickCheck/StateModel.hs

```diff
@@ -178,15 +174,6 @@ instance {-# OVERLAPPING #-} IsPerformResult Void a where
 instance {-# OVERLAPPABLE #-} (PerformResult e a ~ Either e a) => IsPerformResult e a where
   performResultToEither = id
 
--- TODO: maybe it makes sense to write
--- out a long list of these instances
-type family Realized (m :: Type -> Type) a :: Type
```

Comment:
Try something like 
```
{-# DEPRECATED "Providing a Realized instance is not needed anymore" #-}
```
?

Reply:
It would still have to pollute the precondition of RunModel if we want it to be non-breaking and then I don't see the point. Better to rip off the band aid instead of living with our past mistakes now while we still don't have too many users.

Reply:
I gave a quick shot at deprecating `Realized` to check how it would impact hydra code. It works fine by removing everything but the declaration of the type, eg. you don't have to keep it anywhere else and have it pollute the `RunModel` code.  `Lookup`'s signature also changed and it's used in some non-necessary instance method signature.

Reply:
I still don't think we have enough users to worry about keeping this around. Removing it from a code-base should be straightforward enough.